sarif/include/gap/sarif/definitions.hpp

```diff
@@ -0,0 +1,2866 @@
+// Copyright 2024-present, Trail of Bits, Inc.
+
+#pragma once
+
+#include <cstdint>
+#include <forward_list>
+#include <optional>
+#include <string>
+#include <unordered_map>
+
+#include <nlohmann/json.hpp>
+
+namespace gap::sarif::definitions
+{
+    enum class version_enum {
+        k2_1_0,
+    };
+
+    NLOHMANN_JSON_SERIALIZE_ENUM(version_enum, {
+        { version_enum::k2_1_0, "2.1.0" },
+    })
+    struct run_struct;
+    struct external_properties_struct;
+
+    //
+    // Key/value pairs that provide additional information about the object.
+    //
+    struct property_bag_struct {
+        //
+        // A set of distinct strings that provide additional information.
+        //
+        std::optional<std::forward_list<std::string>> tags = std::nullopt;
+
+        std::unordered_map<std::string, nlohmann::json> additional_properties;
+    };
+
+    void to_json(nlohmann::json &, const property_bag_struct &);
```

Comment:
why don't you use `NLOHMANN_DEFINE_TYPE_NON_INTRUSIVE`?

Reply:
A few reasons:
- It doesn't support (de)serializing to/from `std::optional`
- Even if it did, it would probably end up serializing `nullopt` to `null`, whereas I want `nullopt` fields to _not be there at all_
- SARIF specifies that some objects may have additional properties that are not specified, so I still need a field to store those, and that's not supported by the macro